org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/Member.java

```diff
@@ -52,31 +69,50 @@ protected static boolean areSimilarMethods(
 			for (int i = 0; i < params1Length; i++) {
 				String typeErasureParam1Signature = Signature.getTypeErasure(params1[i]);
 				String typeErasureParam2Signature = Signature.getTypeErasure(params2[i]);
+				String simpleName1 =
+					simpleNames1 == null ?
+						Signature.getSimpleName(Signature.toString(typeErasureParam1Signature)) :
+						simpleNames1[i];
+				String simpleName2 = Signature.getSimpleName(Signature.toString(typeErasureParam2Signature));
+				if (!simpleName1.equals(simpleName2)) {
+					return false;
+				}
 				String param1Qualifier = Signature.getSignatureQualifier(typeErasureParam1Signature);
 				String param2Qualifier = Signature.getSignatureQualifier(typeErasureParam2Signature);
-				if (!param1Qualifier.isEmpty() && !param2Qualifier.isEmpty()) {
-					// both qualified -> compare qualified type
-					String qualifiedType1 = Signature.toString(typeErasureParam1Signature);
-					String qualifiedType2 = Signature.toString(typeErasureParam2Signature);
-					if (!Objects.equals(qualifiedType1, qualifiedType2)) {
-						return false;
-					}
-				} else { // at least 1 unqualified name -> only compare simple names
-					String simpleName1 =
-						simpleNames1 == null ?
-							Signature.getSimpleName(Signature.toString(typeErasureParam1Signature)) :
-							simpleNames1[i];
-					String simpleName2 = Signature.getSimpleName(Signature.toString(typeErasureParam2Signature));
-					if (!simpleName1.equals(simpleName2)) {
-						return false;
-					}
+				if (param1Qualifier.isEmpty() || param2Qualifier.isEmpty()
+					|| Objects.equals(param1Qualifier, param2Qualifier)) {
+					continue;
+				}
+				// qualifier can have multiple forms, particularly for nested types:
+				// * mypackage.Outer.Inner.Innest
+				// * Outer.Inner.Innest (if Outer is is imported)
+				// * Inner.Innest (if Inner is imported)
+				// so we compare the suffix
+				if (!(param1Qualifier.endsWith('.' + param2Qualifier) || param2Qualifier.endsWith('.' + param1Qualifier))) {
+					return false;
 				}
 			}
 			return true;
 		}
 	}
 	return false;
 }
+
+/**
+ *
+ * @param candidate The array to test for potential suffix
+ * @param suffix The suffix to look for in candidate array
+ * @return <code>true</code> if last items in <code>candidate</code> are the ones of <code>suffix</code>
+ */
+private static boolean isSuffixOf(String[] candidate, String[] suffix) {
+	if (candidate.length < suffix.length) {
+		return false;
+	}
+	if (suffix.length == 0) {
+		return true;
+	}
+	return Arrays.equals(candidate, candidate.length - suffix.length, candidate.length, suffix, 0, suffix.length);
+}
```

Comment:
once you remove this left-over, we should be ready for merge.

Reply:
Done.

Reply:
thanks.